src/main/java/net/openhft/chronicle/bytes/NativeBytes.java

```diff
@@ -300,6 +302,17 @@ private void resize(final long endOfBuffer)
         }
     }
 
+    @NotNull
+    private BytesStore allocate(long size) {
+        final BytesStore store;
+        try {
+            store = allocateNewByteBufferBackedStore(Maths.toInt32(size));
```

Comment:
Maybe bake the check in allocateNewByteBufferBackedStore

Reply:
The purpose here was to get rid of nested try/catch